extensions/workflowEngineDataSource/WorkflowEngineDataSourceQueuePopulator.js

```diff
@@ -49,55 +52,66 @@ class WorkflowEngineDataSourceQueuePopulator extends QueuePopulatorExtension {
     }
 
     /**
-     * Load a specified child content in the filterDescriptors map do
-     * some sanity checks before
+     * Load a specified child content in the filterDescriptors
+     * map. Perform some sanity checks. Create the entry or update
+     * refcount if already exists
      *
-     * @param {string} name - the ZK child name (must obey syntax wf_id_ver)
-     * @param {object} fd - the filter descriptor object
+     * @note We assume that data sources for a
+     * (workflowId,workflowVersion) pair are identical since versions
+     * are immutable.
+     *
+     * @param {string} name - the ZK child name
+     * @param {object} value - the filter descriptor object
      *
      * @return {object} validationResult - {isValid, message}
      */
-    _loadFilterDescriptor(name, fd) {
-        this.log.info(`WEDSQP: loading new filter for ${name}`, fd);
-        const arr = name.split('_');
-        if (arr[0] !== 'wf') {
-            return {
-                isValid: false,
-                message: `invalid child name: ${arr[0]}`
-            };
-        }
-        if (fd.workflowId === undefined ||
-            fd.workflowVersion === undefined ||
-            fd.type === undefined ||
-            fd.subType === undefined ||
-            fd.nextNodes === undefined) {
+    _loadFilterDescriptor(name, value) {
+        this.log.debug(`WEDSQP: loading ${name}`, value);
+        // sanity checks
+        if (value.workflowId === undefined ||
+            value.workflowVersion === undefined ||
+            value.subType === undefined ||
+            value.nextNodes === undefined) {
             return {
                 isValid: false,
                 message: 'missing some fields'
             };
         }
-        if (fd.subType === this.wed.SUB_TYPE_BASIC) {
-            if (fd.bucket === undefined) {
+        if (value.subType === this.wed.SUB_TYPE_BASIC) {
+            if (value.bucket === undefined) {
                 return {
                     isValid: false,
                     message: 'basic: missing bucket'
                 };
             }
+            if (value.key === undefined) {
+                return {
+                    isValid: false,
+                    message: 'basic: missing key'
+                };
+            }
         }
-        if (fd.workflowId !== arr[1]) {
-            return {
-                isValid: false,
-                message: `non matching workflow id: ${arr[1]}`
-            };
-        }
-        // eslint-disable-next-line
-        if (fd.workflowVersion != arr[2]) {
-            return {
-                isValid: false,
-                message: `non matching workflow version: ${arr[2]}`
-            };
+        // official name of the entry
+        const fdName = this._getHashString(
+            value.workflowId, value.workflowVersion);
+        let fd = this.filterDescriptors[fdName];
+        if (fd === undefined) {
+            fd = value;
+            this.log.info(
```

Comment:
Would it make more sense to have those `WEDSQP` logs turned into debug logs? E.g logging processing on refcount seems a bit too arcane for an info log.

Reply:
This is not a very frequent change and invaluable to understand what is going on.